pkg/cmd/volume/ls.go

```diff
@@ -0,0 +1,230 @@
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"text/tabwriter"
+	"text/template"
+
+	"github.com/containerd/containerd/pkg/progress"
+	types "github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/containerd/nerdctl/pkg/formatter"
+	"github.com/containerd/nerdctl/pkg/inspecttypes/native"
+	"github.com/sirupsen/logrus"
+)
+
+type volumePrintable struct {
+	Driver     string
+	Labels     string
+	Mountpoint string
+	Name       string
+	Scope      string
+	Size       string
+	// TODO: "Links"
+}
+
+func Ls(options *types.VolumeLsCommandOptions) error {
+	if options.Quiet && options.Size {
+		logrus.Warn("cannot use --size and --quiet together, ignoring --size")
+		options.Size = false
+	}
+	labelFilterFuncs, nameFilterFuncs, sizeFilterFuncs, isFilter, err := getVolumeFilterFuncs(options.Filters)
+	if err != nil {
+		return err
+	}
+	if len(sizeFilterFuncs) > 0 && options.Quiet {
+		logrus.Warn("cannot use --filter=size and --quiet together, ignoring --filter=size")
+		sizeFilterFuncs = nil
+	}
+	if len(sizeFilterFuncs) > 0 && !options.Size {
+		logrus.Warn("should use --filter=size and --size together")
+		options.Size = true
+	}
+	w := options.Writer
+	var tmpl *template.Template
+	switch options.Format {
+	case "", "table", "wide":
+		w = tabwriter.NewWriter(options.Writer, 4, 8, 4, ' ', 0)
+		if !options.Quiet {
+			if options.Size {
+				fmt.Fprintln(w, "VOLUME NAME\tDIRECTORY\tSIZE")
+			} else {
+				fmt.Fprintln(w, "VOLUME NAME\tDIRECTORY")
+			}
+		}
+	case "raw":
+		return errors.New("unsupported format: \"raw\"")
+	default:
+		if options.Quiet {
+			return errors.New("format and quiet must not be specified together")
+		}
+		var err error
+		tmpl, err = formatter.ParseTemplate(options.Format)
+		if err != nil {
+			return err
+		}
+	}
+
+	vols, err := GetVolumes(options.Namespace, options.DataRoot, options.Address, options.Size)
+	if err != nil {
+		return err
+	}
+
+	for _, v := range vols {
+		if isFilter && !volumeMatchesFilter(v, labelFilterFuncs, nameFilterFuncs, sizeFilterFuncs) {
+			continue
+		}
+		p := volumePrintable{
+			Driver:     "local",
+			Labels:     "",
+			Mountpoint: v.Mountpoint,
+			Name:       v.Name,
+			Scope:      "local",
+		}
+		if v.Labels != nil {
+			p.Labels = formatter.FormatLabels(*v.Labels)
+		}
+		if options.Size {
+			p.Size = progress.Bytes(v.Size).String()
+		}
+		if tmpl != nil {
+			var b bytes.Buffer
+			if err := tmpl.Execute(&b, p); err != nil {
+				return err
+			}
+			if _, err = fmt.Fprintf(w, b.String()+"\n"); err != nil {
+				return err
+			}
+		} else if options.Quiet {
+			fmt.Fprintln(w, p.Name)
+		} else if options.Size {
+			fmt.Fprintf(w, "%s\t%s\t%s\n", p.Name, p.Mountpoint, p.Size)
+		} else {
+			fmt.Fprintf(w, "%s\t%s\n", p.Name, p.Mountpoint)
+		}
+	}
+	if f, ok := w.(formatter.Flusher); ok {
+		return f.Flush()
+	}
+	return nil
+}
+
+func GetVolumes(ns string, dataRoot string, address string, volumeSize bool) (map[string]native.Volume, error) {
```

Comment:
```suggestion
func Volumes(ns string, dataRoot string, address string, volumeSize bool) (map[string]native.Volume, error) {
```

https://go.dev/doc/effective_go#Getters